### PR TITLE
Update Validation Error Messages for addDisabilitiesRevised

### PR DESCRIPTION
**Pull request by tyler-spangler6**
This updates the validation messages for the addDisabilitiesRevised page based on CAIA review.  These changes are behind the feature flag disability_526_improved_autosuggestions_add_disabilities_page.  

## Summary
Updates were made to the validations for the following situation:
- If the Veterans tries to enter a condition longer than 255 characters 
- If the Veterans tries to enter a condition that was already entered
- If the Veteran tries to submit a claim with no conditions selected
- If the Veteran tries to save an empty condition
- If the Veteran tries to enter more than 100 conditions

## Related issue(s)

- [#468 Update Content for Revised addDisabilities Page](https://app.zenhub.com/workspaces/contention-classification-640a24db5dcd08457cebbc79/issues/gh/department-of-veterans-affairs/vagov-claim-classification/468)
## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Review by lukey-luke — DISMISSED** [review]
Looks good to me!

**Review by tblackwe — DISMISSED** [review]
LGTM

File: src/applications/disability-benefits/all-claims/validations.js [hunk 1]
```diff
@@ -474,7 +480,10 @@ export const validateDisabilityName = (
     item => item === itemLowerCased || sippableId(item) === itemSippableId,
   );
   if (itemCount.length > 1) {
-    err.addError('Please enter a unique condition name');
+    const errorMessage = showRevisedNewDisabilitiesPage()
+      ? "You've already added this condition to your claim"
```
Comment (christinec-fftc) [inline src/applications/disability-benefits/all-claims/validations.js:484]:
```suggestion
      ? "You’ve already added this condition to your claim"
```

see note [here](https://design.va.gov/content-style-guide/punctuation#apostrophes) about curly apostrophes

File: src/applications/disability-benefits/all-claims/validations.js [hunk 2]
```diff
@@ -502,9 +511,10 @@ export const requireDisability = (err, fieldData, formData) => {
  */
 export const limitNewDisabilities = (err, fieldData, formData) => {
   if (formData.newDisabilities?.length > 100) {
-    err.addError(
-      'You have reached the 100 condition limit. If you need to add another condition, you must remove a previously added condition.',
-    );
+    const errorMessage = showRevisedNewDisabilitiesPage()
+      ? "You've added the maximum number of conditions. If you'd like to add another one, you'll need to remove a condition from your claim"
```
Comment (christinec-fftc) [inline src/applications/disability-benefits/all-claims/validations.js:515]:
```suggestion
      ? "You’ve added the maximum number of conditions. If you’d like to add another one, you’ll need to remove a condition from your claim."
```

**Review by ataker — DISMISSED** [review]

**Comment by christinec-fftc** [comment]
Screenshots from testing. Looks good 👍
1. trying to save an empty condition
a. enabled toggle
b. start a new standard claim and fill out required fields until you get to page `disability/file-disability-claim-form-21-526ez/new-disabilities-revised/add`
c. clicked the 'Save' button with an empty input
image

3. trying to submit an empty condition
similar to above, but try to then click the 'Continue' button to go to the next page
image

4. longer than 255 chars
pasted in a condition name that was 257 chars
image

5. add duplicate condition
image

6. entered more than 100 new conditions
image

**Review by christinec-fftc — APPROVED** [review]

**Review by lukey-luke — APPROVED** [review]
LGTM

**Review by ataker — APPROVED** [review]

**Pull request merged**